### ISSUE TITLE
set  "regexFilter"  value, but "maxItemCount" out of action

### ISSUE THREAD

**Issue by mytaia** [jshjohnson/Choices#151]
at line 1089 start 
 
```javascript
   if (this.passedElement.type === 'select-multiple' || this.passedElement.type === 'text') {
      if (this.config.maxItemCount > 0 && this.config.maxItemCount <= this.itemList.children.length) {
        // If there is a max entry limit and we have reached that limit
        // don't update
        canAddItem = false;       // ====== > if canAddItem ==false need then return 
        notice = isType('Function', this.config.maxItemText) ? this.config.maxItemText(this.config.maxItemCount) : this.config.maxItemText;
      }
    }

```

**Comment by jshjohnson** [comment]
Hi @mytaia,

Could you give provide some more info on this please?

Thanks

**Comment by mytaia** [comment]
the code follows 
```javascript

var phones= new Choices(document.getElementById('PHONE'), {
        maxItemCount: 3,
        duplicateItems:false, 
        removeItemButton: true, 
        regexFilter:/^1\d{10}$/
});
```
you can try to input 3 more items  .   



as the code says,if you fill regexFilter ,the maxItemCount is invalid.The problem results that you can add more than three items but you limited it in the code.

**Comment by jshjohnson** [comment]
Hmm, okay. So are you saying that `maxItemCount` is being bypassed if the regex filter passes? I'm confused!

**Comment by jshjohnson** [comment]
Hi @mytaia,

This should now be resolved as of [version 2.8.1](https://github.com/jshjohnson/Choices/releases/tag/v2.8.1).

Thanks
